ext/openssl/openssl.c

```diff
@@ -4228,180 +4228,280 @@ static EVP_PKEY *php_openssl_pkey_init_dh(zval *data, bool *is_private)
 #ifdef HAVE_EVP_PKEY_EC
 #if PHP_OPENSSL_API_VERSION < 0x30000
 static bool php_openssl_pkey_init_legacy_ec(EC_KEY *eckey, zval *data, bool *is_private) {
+	BIGNUM *p = NULL, *a = NULL, *b = NULL, *g_x = NULL, *g_y = NULL, *order = NULL, *cofactor = NULL;
+	BIGNUM *x = NULL, *y = NULL, *d = NULL;
```

Comment:
define those only in the branch where they are needed. No point to have them at the top...

Reply:
> 

If I don't declare the variables initially, a fatal error will occur when releasing the variables at the end. I don't have a better solution now for splitting it into the corresponding branches. Additionally, if an error occurs midway when calling other functions, it will also jump to the place at the end where errors are stored and variables are released. After doing my best to modify this part of the code, if you have a better solution, please help me out. This is my first pull request on PHP source code, and there may be some aspects that I haven't done well, so please point them out and I would be grateful for the feedback.